### ISSUE TITLE
Cannot navigate through or skip the disabled date in Calendar view on DatePicker

### ISSUE THREAD

**Issue by azizoid** [mui/mui-x#6076]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

We can pass the `shouldDisableDate` prop to disable specific date. which will result showing the calendar with some dates disabled. - that is expected behaviour
The problem happens when we want to navigate through the dates using keyboard only. Tab moves us between toolbar, dates grid and action bar. up/right/bottom/left buttons navigates us through the dates. EXCEPT when date is disabled, it does not go through it. if `8th` is disabled, and i want to navigate from `6 -> 7 -> 8 -> 9`. it will stop on `7th`, instead of skipping it and go directly to the `9th`. 
Disaster happens when only `today` is active, then it is selected by default and you cannot navigate anywhere and got kinda stuck

### Expected behavior 🤔

When focus to the Calendar bar with dates, it should focus to the first enabled date, and every further navigation should jump between only enabled dates. 
if `8th` and `11th` are disabled then it should go `6 -> 7 -> 9 -> 10 -> 12 -> 13`

### Steps to reproduce 🕹

Steps:

1. Call DatePicker with some disabled dates using `shouldDisableDate` prop
2. Click the TextField of Datepicker 
3. Press Tab to select the `InputAdornment` open calendar. Press `enter`
4. Press Tab to navigate between the days. 
5. When you reach the disabled date, it will stop and further navigation is impossible

### Context 🔦

We have been told that MUI components fully support a11y, while this example is not acceeible using keybpard only

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by alexfauquette** [comment]
Here is a demo with the fix
https://codesandbox.io/s/eloquent-julien-qfc66h?file=/demo.tsx

It allows ignoring disabled dates with right/left arrow keys

**Comment by azizoid** [comment]
Thank you so much
